presto-docs/src/main/sphinx/functions/array.rst

```diff
@@ -62,6 +62,16 @@ Array Functions
     Sorts and returns the array ``x``. The elements of ``x`` must be orderable.
     Null elements will be placed at the end of the returned array.
 
+.. function:: array_sort(array<T>, function<T,T,bigint>) -> array<T>
```

Comment:
This is a generic, but usually (if not always) what you compare by can be captured more naturally as "comparison function" (similar to Guava `Ordering.onResultOf`). This allows much more succinct expressions.
The examples given below would correspond to the following:
```
SELECT array_sort_by(ARRAY [3, 2, 5, 1, 2], x -> CAST(-x AS BIGINT)); -- [5, 3, 2, 2, 1]
SELECT array_sort_by(ARRAY ['a', 'abcd', 'abc'], x -> length(x)); -- ['a', 'abc', 'abcd']
SELECT array_sort_by(ARRAY ['a', 'abcd', 'abc'], x -> -length(x)); -- ['abcd', 'abc', 'a']
```

(note: this "overload" would not really be an overload, I would propose `array_sort_by` as the name)

@martint  @electrum what do you think?

Reply:
I like this better. It’s easier to write. The example comparator “y - x” is actually incorrect for the MIN_LONG (since the largest negative is one larger than the largest positive due to 2s complement).

Reply:
The result of the lambda could be any comparable type. For example, it could be

s -> reverse(s)